origin_ui/src/components/graphs/Graph.tsx

```diff
@@ -70,14 +70,19 @@ export default function Graph({getData, options, boxProps, drawer}: GraphProps)
 
 
     async function setData() {
-        let response = await getData()
-        _setData(response)
-        setLoading(false)
-        if(response.datasets[0].data.length == 0){
-            let date = new Date(Date.now()).toLocaleTimeString()
-            setError(`No data returned by database as of ${date}; plot will auto-refresh`)
-        } else {
-            setError("")
+        try {
+            let response = await getData()
+            _setData(response)
+            setLoading(false)
+            if(response.datasets[0].data.length == 0){
+                let date = new Date(Date.now()).toLocaleTimeString()
+                setError(`No data returned by database as of ${date}; plot will auto-refresh`)
+            } else {
+                setError("")
+            }
+        } catch (e: any) {
+            let date = new Date(Date.now()).toLocaleString()
+            setError(date + " : " + e.message)
```

Comment:
Do we want to `setLoading(false)` here? It seems that if `getData()` returns error, we will never set loading to false.

Reply:
That makes sense to me! I think one easy improvement can be adding something like "Retry in 1 minute" at the end of the error message so that the user knows it's still updating not just died